File: ios/Reguerta/Reguerta/Domain/Access/AccessResolutionResult.swift
import Foundation

nonisolated enum UnauthorizedReason: Equatable, Sendable {
    case userNotFoundInAuthorizedUsers
    case userAccessRestricted
    case emailVerificationRequired
}

nonisolated enum AccessResolutionResult: Equatable, Sendable {
    case authorized(member: Member, environment: SessionEnvironment)
    case unauthorized(UnauthorizedReason)
}

File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/access/AccessResolutionResult.kt
package com.reguerta.user.domain.access

enum class UnauthorizedReason {
    USER_NOT_FOUND_IN_AUTHORIZED_USERS,
    USER_ACCESS_RESTRICTED,
    EMAIL_NOT_VERIFIED,
}

sealed interface AccessResolutionResult {
    data class Authorized(
        val member: Member,
        val environment: String,
    ) : AccessResolutionResult

    data class Unauthorized(val reason: UnauthorizedReason) : AccessResolutionResult
}
